src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ILScanner.cs

```diff
@@ -275,6 +275,24 @@ public ReadOnlyFieldPolicy GetReadOnlyFieldPolicy()
             return new ScannedReadOnlyPolicy(MarkedNodes);
         }
 
+        public BodyAndFieldSubstitutions GetBodyAndFieldSubstitutions()
+        {
+            Dictionary<MethodDesc, BodySubstitution> bodySubstitutions = [];
+
+            TypeDesc iDynamicInterfaceCastableType = _factory.TypeSystemContext.SystemModule.GetType("System.Runtime.InteropServices", "IDynamicInterfaceCastable");
```

Comment:
GetType has throwIfNotFound arg. You can use it to enable this path only when the type is present in CoreLib, and avoid omit IDynamicInterfaceCastable from Test.CoreLib. I assume that the type exists in Test.Corelib just to make this check happy.

Reply:
`IDynamicInterfaceCastable` exists in CoreLib because TypeCast and CachedInterfaceDispatch call into the interface directly now instead of going through a "support" static class.